test/codeu/chat/server/ConversationAccessServerTest.java

```diff
@@ -0,0 +1,465 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ *  Note: The following names of methods in this tester are tentative. They are just  
+ *  placeholders (if you know a better name for a method, tell me!) and will mirror the 
+ *  corresponding methods as we implement them, so let me know when you push a new commit
+ *  that changes the names of what I have listed here. ~ Sarah Abowitz
+ *
+ *  Replaceable parts:
+ *  - codeu.chat.server.Controller.getPermissionMap(ConversationHeader conversation)
+ *  - codeu.chat.server.Controller.updatePermissionMap(Uuid id, HashMap<Uuid, byte> accessMap)
+ *  - codeu.chat.server.Controller.checkMembership(Uuid id, ConversationHeader conversation)
+ *  - ...server.Controller.demoteOwner(byte access, Uuid target, ConversationHeader conversation)
+ *  - ...server.Controller.addOwner(byte access, Uuid target, ConversationHeader conversation)
+ *  - ...server.Controller.ban(byte access, Uuid target, ConversationHeader conversation)
+ *  - ...server.Controller.addMember(byte access, Uuid target, ConversationHeader conversation)
+ *  - codeu.chat.server.View.getPermissionMapAsString(ConversationHeader conversation)
+ *    
+ */
+
+
+package codeu.chat.server;
+
+import java.util.HashMap;
+
+import static org.junit.Assert.*;
+import org.junit.Test;
+import org.junit.Before;
+
+import codeu.chat.common.BasicController;
+import codeu.chat.common.BasicView;
+import codeu.chat.common.ConversationHeader;
+import codeu.chat.common.Message;
+import codeu.chat.common.User;
+import codeu.chat.util.Uuid;
+
+public final class ConversationAccessServerTest {
+
+  private Model model;
+  private BasicController controller;
+  private BasicView view;
+
+  @Before
+  public void doBefore() {
+    model = new Model();
+    controller = new Controller(Uuid.NULL, model);
+    view = new View(model);
+  }
+
+  @Test
+  public void testAddConversation() {
+
+    final User user = controller.newUser("user");
+
+    assertFalse(
```

Comment:
Beyond asserting that the User is not null, it would also be good to make some assertions about the user's properties as well - i.e. is the user's name set as you expect? Does it have a valid UUid? These types of assertions are good because they test even more of the controller's logic.